### PR TITLE
adds modifyBabelConfig

### LINKED ISSUE #134: Refactor how Babel configuration is loaded in
This came up as part of the work on #115 . Right now, the way the Babel config is loaded for Jest is by running the `modifyWebpackConfig()` callback and grabbing it from there, but this overloads that method a bit. Now that there are 2 consumers for the Babel configuration (Webpack and Jest), it would be cleaner to pull that out.

Here's how it could work:
- Userland can optionally define a `.babelrc` file if they have any custom plugins or presets that they would like to use. (We should also check `package.json` for the configuration.)
- We'll respect the `BABEL_ENV` setting to pull in the appropriate dev/prod/test babel configuration.
- Merge it in with kyt's base babel configuration.

For most use cases, a separate babelrc file in userland won't be needed, but in the cases where it is, it will be easier to configure than having to do so programmatically in the `modifyWebpackConfig()` callback which requires sniffing for the `babel-loader`.

### PR DESCRIPTION
**Pull request by jaredmcdonald**
fixes #134 

adds a `modifyBabelConfig` callback in `kyt.config.js`

### CODE REVIEW

File: config/babel.js [hunk 1]
```diff
@@ -26,5 +30,16 @@ module.exports = (options) => {
...
   Object.keys(babelrc.env || {}).forEach(env => resolvePluginsPresets(babelrc.env[env]));
 
-  return babelrc;
+  // modify via userland kytConfig's `modifyBabelConfig`
+  try {
+    const mergedBabelrc = config.modifyBabelConfig(babelrc, options);
+    if (config.debug) {
+      logger.debug('Merged babel configuration:', mergedBabelrc);
+    }
+    return mergedBabelrc;
```
Comment (tizmagik) [inline config/babel.js]: This is a misnomer right? There's no actual merging going on.

File: utils/jest/__tests__/preprocessor.test.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+const fakeBabelResult = {};
+
+jest.setMock('../../../config/babel', jest.fn(() => fakeBabelResult));
+const babel = require('../../../config/babel');
+
+describe('preprocessor', () => {
+  it('calls the babel utility and babel-jest\'s createTransformer', () => {
+    const preprocessor = require('../preprocessor'); // eslint-disable-line global-require
+    expect(preprocessor).toBe(fakeTransformer);
```
Comment (tizmagik) [inline utils/jest/__tests__/preprocessor.test.js]: Isn't this just testing Jest's ability to mock?

File: utils/jest/__tests__/preprocessor.test.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    expect(babel.mock.calls[0][0]).toEqual({
+      type: 'test',
+      environment: 'test',
+    });
+    expect(babelJest.createTransformer).toBeCalledWith(fakeBabelResult);
+  });
+});
```
Comment (tizmagik) [inline utils/jest/__tests__/preprocessor.test.js]: Thanks for adding this test!! 🙏

File: config/babel.js (hunk 1 above)
Comment (jaredmcdonald) [inline config/babel.js]: yeah, maybe `modifiedBabelConfig` would be more accurate?

Comment (tizmagik) [inline config/babel.js]: Yea that sounds good

File: utils/jest/__tests__/preprocessor.test.js (hunk 2 above)
Comment (jaredmcdonald) [inline utils/jest/__tests__/preprocessor.test.js]: although, wait, I added it to verify that `preprocessor` exports the result of `createTransformer`

Comment (tizmagik) [inline utils/jest/__tests__/preprocessor.test.js]: Ah ok gotcha!

File: config/babel.js (hunk 1 above)
Comment (jaredmcdonald) [inline config/babel.js]: @tizmagik - done

**Comment by delambo** [comment]
Can we pause this? I'd like to see how #250 feels first.

**Comment by jaredmcdonald** [comment]
sounds good to me @delambo--I'll close the PR for now, can always reopen

**Pull request closed without merging**